Add 'coordinating multiple agents' (#2210)
buildkite/docs: pages/tutorials/parallel_builds.md

buildkite-agent start --tags queue=deploy
```

#### Coordinating multiple agents

>🛠️ Experimental feature
> To use it, set <code>experiment="agent-api"</code> in your <a href="/docs/agent/v3/configuration#experiment"> agent configuration</a>.

> > This also requires Agent v3.47.0 or later.

Reviewer: I don't feel too strongly about this, but I think you can ditch the "also."

```suggestion
> This requires Agent v3.47.0 or later.
```

> Multiple agents on a single host can sometimes interfere with one another. For example, a pipeline might contain commands like `docker prune` or `apt upgrade`, but these commands fail if another job is running the same command at the same time.

Reviewer: General tightening:

```suggestion
Multiple agents on a single host can sometimes interfere with one another. For example, a pipeline might contain commands like `docker prune` or `apt upgrade`, but these commands fail if another job runs the same command at the same time.
```

You could also consider changing "at the same time" to "simultaneously," but I'm not sure it makes it any clearer.

To coordinate access to shared resources on the same host, you can use agent locks. Locking is advisory (nothing prevents a buggy command from ignoring a lock), but it can help avoid multiple agents interfering with each other.

> Here's how you could use locks to make sure a command is run by only one agent at a time:

Reviewer: Where would these commands be defined? I assume in a script? Could we mention something about that in this lead in? Thinking something like:

```suggestion
Here's how you could use locks in a script to make sure a command is run by only one agent at a time:
```
